packages/engine/lib/execution/src/package/comms.rs

```diff
@@ -0,0 +1,67 @@
+use async_trait::async_trait;
+use stateful::field::PackageId;
+use tracing::Instrument;
+
+use crate::{
+    package::{PackageTask, PackageType},
+    task::{ActiveTask, Task, TaskSharedStore},
+    Result,
+};
+
+/// Temporary trait to proceed with moving the package system into this crate
+#[async_trait]
+pub trait Comms: Send + Sync + 'static {
+    type ActiveTask: ActiveTask;
+
+    /// Takes a given [`Task`] object, and starts its execution on the [`worker_pool`], returning an
+    /// [`ActiveTask`] to track its progress.
+    ///
+    /// [`worker_pool`]: crate::worker_pool
+    async fn new_task(
+        &self,
+        package_id: PackageId,
+        task: PackageTask,
+        shared_store: TaskSharedStore,
+    ) -> Result<Self::ActiveTask>;
+}
+
+pub struct PackageComms<C> {
+    comms: C,
+    package_id: PackageId,
+    _package_type: PackageType, // TODO: unused, remove?
+}
+
+impl<C> PackageComms<C> {
+    pub fn new(comms: C, package_id: PackageId, package_type: PackageType) -> Self {
+        Self {
+            comms,
+            package_id,
+            _package_type: package_type,
+        }
+    }
+}
+
+impl<C: Comms> PackageComms<C> {
+    pub async fn new_task(
+        &self,
+        task: PackageTask,
+        shared_store: TaskSharedStore,
+    ) -> Result<C::ActiveTask> {
+        let task_name = task.name();
+
+        self.comms
+            .new_task(self.package_id, task, shared_store)
+            .instrument(tracing::debug_span!("Task", name = task_name))
+            .await
+    }
+
+    //     // TODO: UNUSED: Needs triage
```

Comment:
We should probably decide what to do with this in this PR.

The intention is that Packages should be able to create and remove agents as well.

I think considering the implementation is already there we should probably leave it in and just allow unused code in clippy until we add more packages

Reply:
Yes, let's keep them: 1c02fb52eef8296de3ea52faa98adaf9b357ff1e